dask/dataframe/io/json.py

```diff
@@ -192,24 +203,62 @@ def read_json(
             "input (orient='records', lines=True)."
         )
     storage_options = storage_options or {}
+    if include_path_column is True:
+        include_path_column = "path"
+
+    if path_converter is None:
+        path_converter = lambda x: x
+
     if blocksize:
-        first, chunks = read_bytes(
+        b_out = read_bytes(
             url_path,
             b"\n",
             blocksize=blocksize,
             sample=sample,
             compression=compression,
+            include_path=include_path_column,
             **storage_options,
         )
-        chunks = list(flatten(chunks))
+        if include_path_column:
+            first, chunks, paths = b_out
+            first_path = path_converter(paths[0])
+            path_dtype = pd.CategoricalDtype(path_converter(p) for p in paths)
```

Comment:
I'm curious why `pd.CategoricalDtype` is needed here. Is `object` not sufficient here for some reason? 

What `dtype` is used when `include_path_column=True` in `read_csv`?

Reply:
`pd.CategoricalDtype` isn't needed here, but `read_csv` uses a categorical type to save memory. See the discussion here: https://github.com/dask/dask/pull/3908#issuecomment-416943078

Reply:
I'll add a test to enforce this, in parallel with `test_csv.py::test_read_csv_include_path_column_is_dtype_category`

Reply:
Okay, I hadn't realized we were doing this in `read_csv`. Let's match here as you've already done